google/api_core/timeout.py

```diff
@@ -222,3 +224,73 @@ def __str__(self):
                 self._initial, self._maximum, self._multiplier, self._deadline
             )
         )
+
+
+def _logical_call_timeout_generator(overall_timeout):
+    """A generator that yields logical call timeout values.
+
+    Args:
+        overall_timeout (float): The overall timeout of the logical call.
+            Each attempt uses the time remaining until the deadline, calculated
+            using the overall timeout from the start of the logical call.
+
+    Yields:
+        float: A timeout value.
+    """
+    deadline_datetime = datetime_helpers.utcnow() + datetime.timedelta(
+        seconds=overall_timeout
+    )
+    timeout = overall_timeout
+    while True:
+        yield timeout
+        now = datetime_helpers.utcnow()
+        timeout = float((deadline_datetime - now).seconds)
+
+
+@six.python_2_unicode_compatible
+class LogicalCallTimeout(object):
+    """A decorator that adds a logical call timeout argument.
+
+    If called multiple times, each time this decorator will calculate a new
+    timeout parameter based on the time remaining in the overall timeout.
+
+    For example
+
+    .. code-block:: python
```

Comment:
This block needs to be populated, or it will break Sphinx.  E.g.:

```python
   logical_timeout = LogicalCallTimeout(100)

   @logical_timeout
   def make_api_call(foo, bar=None):
       ...
```

Reply:
Ah ok, yeah this was a copy paste misstep from `ExponentialTimeout` declared above - it is also missing code block contents. I will add something.

Reply:
Done.